CRM/Contract/Form/Modify.php

```diff
@@ -11,6 +11,13 @@ class CRM_Contract_Form_Modify extends CRM_Core_Form{
 
   function preProcess(){
 
+    // If we requested a contract file download
+    $download = CRM_Utils_Request::retrieve('ct_dl', 'String', CRM_Core_DAO::$_nullObject, FALSE, '', 'GET');
+    if (!empty($download)) {
+      CRM_Contract_Utils::downloadContractFile($download);
```

Comment:
That's weird, that works for me. The first parameter is the name, you've seen that, right? Maybe it's a browser issue? What are you using?